### PR TITLE
[Notifier] Update for V3 Engagespot API

### PR DESCRIPTION
**Pull request by dave-simshvac**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Fixes a bug in the Engagespot notifier for the v3 API, two different API keys are required, and the response returned is an HTTP 202. Further, the response received from Engagespot API is not JSON

### CODE REVIEW

**Comment by fabpot** [comment]
@danut007ro Can you have a look at this change?

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
I guess we'll have to trust you on this one @dave-simshvac 
Please mind the psalm report, it found an issue.

Friendly ping @danut007ro in case you can have a look before we merge.

File: src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotOptions.php
Comment (stof) [inline src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotOptions.php:43]: Removing a method in a patch release is a no-go

File: src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php
Comment (stof) [inline src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php:35]: Replacing the second argument is a BC break (and a silent one here as it will start using the campaign name as api secret)

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by butitworksonmymachine** [comment]
hi, I'm following this thread for sometime. Any new updates?

**Comment by fabpot** [comment]
Some comments are not taken into account yet, @butitworksonmymachine Do you want to take over an finish this PR?

File: src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php
Comment (nicolas-grekas) [reply src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php:35]: please use `#[\SensitiveParameter]`

**Comment by fabpot** [comment]
I'm closing this PR as there is no more activity. Feel free to reopen when ready.

File: src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php
Comment (meanands) [reply src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php:35]: Hi, I'm the founder & maintainer of Engagespot. I can confirm that nobody is using version v2 of Engagespot Symfony integration. So isn't it is safe to do this change there is no concern of backward compatibility?

Comment (fabpot) [reply src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransport.php:35]: Packagist: https://packagist.org/packages/symfony/engagespot-notifier/stats
@anandrmedia What if someone starts using the integration via Symfony 6.0 and then move on to 6.1?
In any case, this PR is close as there was no more activity. Someone would need to take over and create a new one to move things forward.

**Pull request closed without merging**
